### PR TITLE
Eliminate GraphQL config elements instantiation on the fly

### PR DESCRIPTION
**Pull request by akaplya**
Eliminate GraphQL config elements instantiation on the fly

### CODE REVIEW

**Comment by larsroettig** [comment]
@akaplyaI have a pr for a similar thing https://github.com/magento/graphql-ce/pull/314

**Comment by paliarush** [comment]
Proposal looks good, please test performance to make sure this change gives us any improvement.

File: design-documents/graph-ql/framework/eliminate-config-elements-instantiation-on-the-fly.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+            <argument name="serializer" xsi:type="string">Magento\Framework\GraphQl\Config\Data\Serializer</argument>
+        </arguments>
+    </virtualType>
+```
+In a fact, Magento knows all config data objects.
+Magento does not use external resources as a configuration source.
+So we can create a lightweight serializer which may utilize 
+[serialize](https://secure.php.net/manual/en/function.serialize.php)
+/*
+[unserialize](https://secure.php.net/manual/en/function.unserialize.php) functions with all known GraphQL configuration data objects mentioned in *`allowed_classes` option.
```
Comment (buskamuza) [inline design-documents/graph-ql/framework/eliminate-config-elements-instantiation-on-the-fly.md]: We can't use `unserialize` from security perspective.

**Comment by paliarush** [comment]
Closing due to high risk/reward ratio. Proposed approach can be reconsidered once again in the future.

**Pull request closed without merging**